frontend/module/yace-editor.js

```diff
@@ -77,6 +77,7 @@ export default class Yace {
 
     if (
       (value === this.value || value == null) &&
+      Object.keys(this.errorLines).length === 0 &&
```

Comment:
Isn't this already handled in the next line of the condition? Or is my logic failing me? Either way, I don't really understand this condition. Are these two lines really what is needed to check if an object is empty? Isn't that what this condition is saying? "if the value has not changed or it is null AND there are no errors to highlight then return"?

Even so, this logic is "if foo AND (foo OR bar)" (reordering some things). If foo is true, then bar doesn't matter. If foo is false, then bar doesnt matter. Yeah, something seems off here, but maybe that's me missing something.

Reply:
the commit message describes the error case - maybe you have a better idea on how to fix it?

essentially i want to say: if the new `value` and the `errLines` are the same then don't do anything....

the issue I had was (and I really need tests):

```
x := "foo"  // line 1: 'x' declared but not used
print y     // line 2: unknown variable name 'y'
```
Fixed as
```
x := "foo"  // 💣 error remained: line 1: 'x' declared but not used
print x     // no error
 ```

Reply:
ok, sorry for the late response. The difference is `this.errorLines` vs `errorLines`.

Reply:
This means:
If we haven't had any errorHighlights (`this.errorLines`) and we are also not any new errors (`errorlines`).

Reply:
ok. sorry I didn't see the difference between those two. :moneybag: